- change mouse input from int to float and adjust games accordingly.

mjr4077au opened the pull request:
When I was looking at all the controller code, I thought the mouse movement code would be better off being a float for more precision. If we're going to implicitely convert it to float by dividing it by float-type CVARSs, might as well maintain it.

Overall the game code has been simplified in the process as rather than dividing mousex/y by 32/64 respectively, the appropriate scaling is done in `GetMouseDelta()`.

* Calculate game-side mousex/mousey divisions into the calculations performed in `InputState::GetMouseDelta()`.
* Standard mouse forward/side movement speeds in Exhumed & SW with that of other games.
* Remove `strafeyaw` code from Duke/Exhumed/RR as it's not necessary and was leading to situations where the player would continually keep moving sideways even without input.
* Change mouse forward/side velocities to -= current value as is done with controller input and the player's angle/aim velocities.

sinisterseed commented:
> * Remove `strafeyaw` code from Duke/Exhumed/RR as it's not necessary and was leading to situations where the player would continually keep moving sideways even without input.

This really caught my attention. I always felt there was an awkward sidestepping in Duke after running, but assumed that's just the way the game was meant to be or Build shenanigans. Yet, that's actually not the case. 

Very good that you managed to find this and nuke it into oblivion, most useless quirk ever, and made full stops after running around odd at times.

mjr4077au commented:
@sinisterseed with that `strafeyaw` code, please note that's only when you use hold the strafe key down (RAlt by default) and move the mouse sideways for strafing. If you have an issue with strafing just using the keyboard (like in a WSAD configuration), you will find it's still going to be the same.

mjr4077au commented:
As a side note, force-pushed to fix an issue I saw with `in_mousesmoothing` in that the 3.f divider that was used in the `finput` vector wasn't applying to it, therefore angle movement was insanely fast for then player and not very smooth.

Also changed some of the code I changed in `inputstate.cpp` to not use 'magic number constants' and use the original 16.f, but dividing where the divisions that were done game-side have been removed. Should hopefully improve clarity of the code at the expense of a few more calculations :)

coelckers commented:
Ok - but why not go all the way and turn those q16 variables into floats as well - it'd remove a large amount of these ugly conversion calls. ;)

mjr4077au commented:
@coelckers one of the reasons that the fixed point math was used was that the angle can only be -2047 to 2047 (0x7FFFFFFF) and I don't think you can wrap around a float with an `& 2047` at the end. The other reason is that the fixed point math is now fully exposed to CON scripting so it kind of has to stay for Duke 3D, meaning it probably should just stay for the other games for consistency.
